arch/xtensa/src/esp32s2/esp32s2_allocateheap.c

```diff
@@ -59,13 +59,12 @@
 
 void up_allocate_heap(void **heap_start, size_t *heap_size)
 {
-  extern uint32_t *_dram0_rtos_reserved_start;
+  extern char _dram0_rtos_reserved_start[];
```

Comment:
```suggestion
  extern uint8_t _dram0_rtos_reserved_start[];
```
CI error:
```
chip/esp32s2_allocateheap.c: In function 'up_allocate_heap':
Error: chip/esp32s2_allocateheap.c:67:43: error: invalid operands to binary - (have 'char *' and 'uint8_t *' {aka 'unsigned char *'})
   *heap_size = _dram0_rtos_reserved_start - _sheap;
                ~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
make[1]: *** [Makefile:120: esp32s2_allocateheap.o] Error 1
make[1]: Target 'libarch.a' not remade because of errors.
make: *** [tools/LibTargets.mk:152: arch/xtensa/src/libarch.a] Error 2
make: Target 'all' not remade because of errors.
```

Reply:
Done.